Content.Server/Medical/CrewMonitoring/CrewMonitoringServerSystem.cs

```diff
@@ -0,0 +1,169 @@
+﻿using Content.Server.Destructible;
+using Content.Server.DeviceNetwork;
+using Content.Server.DeviceNetwork.Components;
+using Content.Server.DeviceNetwork.Systems;
+using Content.Server.Medical.SuitSensors;
+using Content.Server.Power.Components;
+using Content.Shared.Medical.SuitSensor;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Medical.CrewMonitoring;
+
+public sealed class CrewMonitoringServerSystem : EntitySystem
+{
+    [Dependency] private readonly SuitSensorSystem _sensors = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly DeviceNetworkSystem _deviceNetworkSystem = default!;
+
+    private const float UpdateRate = 3f;
+    private float _updateDif;
+    private EntityUid? _activeServer;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<CrewMonitoringServerComponent, ComponentRemove>(OnRemove);
+        SubscribeLocalEvent<CrewMonitoringServerComponent, DeviceNetworkPacketEvent>(OnPacketReceived);
+        SubscribeLocalEvent<CrewMonitoringServerComponent, PowerChangedEvent>(OnPowerChanged);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        // check update rate
+        _updateDif += frameTime;
+        if (_updateDif < UpdateRate)
+            return;
+        _updateDif = 0f;
+
+        if (_activeServer == null)
+        {
+            var servers = EntityManager.EntityQuery<CrewMonitoringServerComponent>().GetEnumerator();
+            CrewMonitoringServerComponent? server = null;
+
+            //Retrieve the first available server
+            while (servers.MoveNext())
+            {
+                if (servers.Current.Available)
+                {
+                    server = servers.Current;
+                }
+            }
+
+            servers.Dispose();
+
+            if (server == null)
+                return;
+
+            _activeServer = server.Owner;
+            _deviceNetworkSystem.ConnectDevice(server.Owner);
+        }
+
+        if (_activeServer == null)
+            return;
```

Comment:
Making this a singleton seems bad.

Reply:
I'll most likely make this one server per station/grid so it wouldn't be a singleton anymore at that point.